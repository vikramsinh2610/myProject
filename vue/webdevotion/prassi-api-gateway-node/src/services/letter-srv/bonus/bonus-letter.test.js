const BonusLetter = require('./bonus-letter');

describe('BonusLetter', () => {
  test('should have these properties', () => {
    const letter = new BonusLetter({
      _id: '01-2020',
      status: 'wip',
      promoterId: 'f34g3df-f3c3x2ed-d232r4g5h5-d21s',
      promoterSerialNumber: 'AZ123456',
      promoterDisplayName: 'Pippo De Peppis',
      type: 'rappel',
      paymentFrequency: 'one-time',
      description: 'Bla bla bla',
      signatureDate: new Date(),
      attachmentIds: ['a', 'b'],
      conditionedBonuses: [{}],
      guaranteedBonuses: [{}],
      fromProductivePeriodYear: 2018,
      fromProductivePeriodMonth: 1,
      toProductivePeriodYear: 2019,
      toProductivePeriodMonth: 1,
    });
    expect(letter).toHaveProperty('_id');
    expect(letter).toHaveProperty('status');
    expect(letter).toHaveProperty('promoterId');
    expect(letter).toHaveProperty('promoterSerialNumber');
    expect(letter).toHaveProperty('promoterDisplayName');
    expect(letter).toHaveProperty('paymentFrequency');
    expect(letter).toHaveProperty('type');
    expect(letter).toHaveProperty('description');
    expect(letter).toHaveProperty('signatureDate');
    expect(letter).toHaveProperty('attachmentIds');
    expect(letter).toHaveProperty('cumulateConditionedBonuses');
    expect(letter).toHaveProperty('conditionedBonuses');
    expect(letter).toHaveProperty('guaranteedBonuses');
    expect(letter).toHaveProperty('fromProductivePeriodYear');
    expect(letter).toHaveProperty('fromProductivePeriodMonth');
    expect(letter).toHaveProperty('toProductivePeriodYear');
    expect(letter).toHaveProperty('toProductivePeriodMonth');
  });

  test('should return it s JSON Schema', () => {
    expect(BonusLetter.getJSONSchema()).toMatchSnapshot();
  });
});
