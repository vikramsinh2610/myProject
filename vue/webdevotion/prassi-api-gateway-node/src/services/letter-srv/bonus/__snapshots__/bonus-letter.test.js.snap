// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BonusLetter should return it s JSON Schema 1`] = `
Object {
  "description": "Bonus Letter",
  "properties": Object {
    "_id": Object {
      "type": "string",
    },
    "attachmentIds": Object {
      "description": "Attachements ID",
      "items": Object {
        "type": "string",
      },
      "type": "array",
    },
    "conditionedBonuses": Object {
      "items": Object {
        "description": "Conditioned Bonus",
        "properties": Object {
          "amount": Object {
            "type": "integer",
          },
          "conditions": Object {
            "items": Object {
              "properties": Object {
                "targets": Object {
                  "items": Object {
                    "properties": Object {
                      "kpi": Object {
                        "properties": Object {
                          "_id": Object {
                            "type": "string",
                          },
                          "options": Object {
                            "additionalProperties": true,
                            "type": "object",
                          },
                          "type": Object {
                            "enum": Array [
                              "number",
                              "currency",
                              "percentage",
                            ],
                            "type": "string",
                          },
                          "value": Object {
                            "type": "integer",
                          },
                        },
                        "required": Array [
                          "_id",
                        ],
                        "type": "object",
                      },
                      "targetValue": Object {
                        "type": "integer",
                      },
                      "weightPercentage": Object {
                        "maximum": 10000,
                        "minimum": 0,
                        "type": "integer",
                      },
                    },
                    "required": Array [
                      "kpi",
                      "targetValue",
                      "weightPercentage",
                    ],
                    "type": "object",
                  },
                  "type": "array",
                },
              },
              "required": Array [
                "targets",
              ],
              "type": "object",
            },
            "type": "array",
          },
          "maxRecoveryPercentage": Object {
            "maximum": 10000,
            "minimum": 0,
            "type": "integer",
          },
          "paymentTime": Object {
            "enum": Array [
              "prepayment",
              "payment-delayed",
            ],
            "type": "string",
          },
          "productivePeriodPaymentDelay": Object {
            "maximum": 3,
            "minimum": 0,
            "type": "integer",
          },
          "type": Object {
            "enum": Array [
              "conditioned",
            ],
            "type": "string",
          },
        },
        "required": Array [
          "amount",
          "type",
          "paymentTime",
          "conditions",
        ],
        "type": "object",
      },
      "type": "array",
    },
    "cumulateConditionedBonuses": Object {
      "default": false,
      "type": "boolean",
    },
    "description": Object {
      "type": "string",
    },
    "didActiveDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "didCreateDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "didDeleteDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "didExpireDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "fromProductivePeriodMonth": Object {
      "maximum": 12,
      "minimum": 1,
      "type": "integer",
    },
    "fromProductivePeriodYear": Object {
      "maximum": 2100,
      "minimum": 2010,
      "type": "integer",
    },
    "guaranteedBonuses": Object {
      "items": Object {
        "description": "Guaranteed Bonus",
        "properties": Object {
          "absorbability": Object {
            "properties": Object {
              "directProductionPercentage": Object {
                "description": "Direct production percentage, with 2 decimals as integer",
                "maximum": 10000,
                "minimum": 0,
                "type": "integer",
              },
              "indirectProductionPercentage": Object {
                "description": "Indirect production percentage, with 2 decimals as integer",
                "maximum": 10000,
                "minimum": 0,
                "type": "integer",
              },
            },
            "required": Array [
              "directProductionPercentage",
              "indirectProductionPercentage",
            ],
            "type": "object",
          },
          "amount": Object {
            "type": "integer",
          },
          "paymentTime": Object {
            "enum": Array [
              "prepayment",
              "payment-delayed",
            ],
            "type": "string",
          },
          "type": Object {
            "enum": Array [
              "guaranteed",
            ],
            "type": "string",
          },
        },
        "required": Array [
          "amount",
          "type",
          "paymentTime",
          "absorbability",
        ],
        "type": "object",
      },
      "type": "array",
    },
    "guaranteedVariableBonuses": Object {
      "items": Object {
        "description": "Guaranteed Variable Bonus",
        "properties": Object {
          "absorbability": Object {
            "properties": Object {
              "directProductionPercentage": Object {
                "description": "Direct production percentage, with 2 decimals as integer",
                "maximum": 10000,
                "minimum": 0,
                "type": "integer",
              },
              "indirectProductionPercentage": Object {
                "description": "Indirect production percentage, with 2 decimals as integer",
                "maximum": 10000,
                "minimum": 0,
                "type": "integer",
              },
            },
            "required": Array [
              "directProductionPercentage",
              "indirectProductionPercentage",
            ],
            "type": "object",
          },
          "amount": Object {
            "type": "integer",
          },
          "paymentTime": Object {
            "enum": Array [
              "prepayment",
              "payment-delayed",
            ],
            "type": "string",
          },
          "type": Object {
            "enum": Array [
              "guaranteed",
            ],
            "type": "string",
          },
          "variableBonus": Object {
            "items": Object {
              "properties": Object {
                "adjustedPercentage": Object {
                  "minimum": 0,
                  "type": "integer",
                },
                "formula": Object {
                  "type": "string",
                },
              },
              "type": "object",
            },
            "minItems": 1,
            "type": "array",
          },
        },
        "required": Array [
          "amount",
          "type",
          "paymentTime",
        ],
        "type": "object",
      },
      "type": "array",
    },
    "invoiceDescription": Object {
      "type": "string",
    },
    "paymentFrequency": Object {
      "enum": Array [
        "monthly",
        "one-time",
      ],
      "type": "string",
    },
    "promoterDisplayName": Object {
      "description": "Promoter full name",
      "type": "string",
    },
    "promoterId": Object {
      "description": "Promoter ID",
      "type": "string",
    },
    "promoterSerialNumber": Object {
      "description": "Promoter Serial Number",
      "type": "string",
    },
    "signatureDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "status": Object {
      "description": "Status of the letter",
      "enum": Array [
        "wip",
        "active",
        "inactive",
        "expired",
        "deleted",
      ],
      "type": "string",
    },
    "toProductivePeriodMonth": Object {
      "maximum": 12,
      "minimum": 1,
      "type": "integer",
    },
    "toProductivePeriodYear": Object {
      "maximum": 2100,
      "minimum": 2010,
      "type": "integer",
    },
    "type": Object {
      "enum": Array [
        "none",
        "job",
        "commissioning-pa",
        "commissioning-pas",
        "bonus-pa",
        "bonus-pas",
        "rappel-pa",
        "rappel-pa-2021",
        "rappel-pas",
        "rappel",
        "rappel-2",
        "rappel-3",
        "rappel-5",
        "welcome-bonus",
        "guaranteed",
        "guaranteed-variable",
        "guaranteed-variable-minimum",
        "guaranteed-with-bonus",
        "guaranteed-with-bonus-prepaid",
        "management-fee",
        "privacy",
      ],
      "type": "string",
    },
    "willExpireDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
  },
  "required": Array [
    "_id",
    "status",
    "promoterId",
    "promoterSerialNumber",
    "promoterDisplayName",
    "type",
    "description",
    "signatureDate",
    "attachmentIds",
    "guaranteedBonuses",
    "conditionedBonuses",
    "paymentFrequency",
  ],
  "type": "object",
}
`;
